cli/ballerina-packerina/src/test/java/org/ballerinalang/packerina/cmd/DependencyScopeTest.java

```diff
@@ -0,0 +1,133 @@
+import static org.wso2.ballerinalang.compiler.util.ProjectDirConstants.BLANG_COMPILED_PKG_BINARY_EXT;
+
+/**
+ * Dependency scope tests.
+ *
+ * @since 1.2.x
+ */
+public class DependencyScopeTest extends CommandTest {
+
+    private Path testResources;
+    private URI testResourcesURI;
+
+    @BeforeClass
+    public void setup() throws IOException {
+        super.setup();
+        try {
+            this.testResources = super.tmpDir.resolve("scope-test-resources");
+            testResourcesURI = getClass().getClassLoader().getResource("test-resources").toURI();
+            Path storedJarDependencyProject = Paths.get(testResourcesURI).resolve("stored-jar-dependency-project");
+            Files.walkFileTree(storedJarDependencyProject, new BuildCommandTest.Copy(storedJarDependencyProject,
+                    this.testResources));
+        } catch (URISyntaxException e) {
+            Assert.fail("error loading resources");
+        }
+    }
+
+    @Test(description = "Test if platform libs are packed with the balo based on the scope.")
+    public void testBuildCommandWithStoredJarDependency() throws IOException {
+        String baloFileName = "mymodule-" + ProgramFileConstants.IMPLEMENTATION_VERSION + "-java8-0.1.0"
+                + BLANG_COMPILED_PKG_BINARY_EXT;
+        Path balo = this.testResources.resolve(ProjectDirConstants.TARGET_DIR_NAME).
+                resolve(ProjectDirConstants.TARGET_BALO_DIRECTORY);
+        File baloFile = new File(balo.toString() + File.separator + baloFileName);
+        File baloZipFile = new File(balo.toString() + File.separator +
+                baloFileName.concat(".zip"));
+        Path scopeToml = Paths.get(testResourcesURI).resolve("scope-toml");
+        String storedJarPath = "platform-libs/storedJar.jar";
+
+        // Build the project
+        String[] compileArgs = {"--all", "--skip-tests"};
+        BuildCommand buildCommand = new BuildCommand(this.testResources, printStream, printStream,
+                false, true);
+        new CommandLine(buildCommand).parse(compileArgs);
+        // default scope
+        buildCommand.execute();
+        Assert.assertTrue(Files.exists(balo), "Check if balo directory exists");
+
+        // Check whether dependency jars getting packed to balo
+        Assert.assertTrue(renameFile(baloFile, baloZipFile));
+        Assert.assertTrue(isJarExists(baloZipFile, storedJarPath));
+
+        // package scope
+        copy(scopeToml.resolve("case1").resolve(ProjectDirConstants.MANIFEST_FILE_NAME),
+                this.testResources.resolve(ProjectDirConstants.MANIFEST_FILE_NAME));
+        buildCommand.execute();
+        Assert.assertTrue(Files.exists(balo), "Check if balo directory exists");
+
+        // Check whether dependency jars getting packed to balo
+        Assert.assertTrue(renameFile(baloFile, baloZipFile));
+        Assert.assertTrue(isJarExists(baloZipFile, storedJarPath));
+
+        // package scope
```

Comment:
When checking the Ballerina.toml file, it looks like "case2" should be the "compile" scope.